diff -ruN '--exclude=target' '--exclude=Cargo.lock' '--exclude=.cargo-checksum.json' '--exclude=.cargo-ok' a/Cargo.toml b/Cargo.toml
--- a/Cargo.toml	2006-07-24 09:21:28.000000000 +0800
+++ b/Cargo.toml	2026-09-29 02:32:19.804165838 +0800
@@ -72,7 +72,6 @@
 [dependencies.curve25519-dalek]
 version = "4"
 features = [
-    "alloc",
     "zeroize",
 ]
 default-features = false
@@ -101,7 +100,6 @@
 version = "1.5"
 features = [
     "zeroize_derive",
-    "alloc",
 ]
 default-features = false
 
diff -ruN '--exclude=target' '--exclude=Cargo.lock' '--exclude=.cargo-checksum.json' '--exclude=.cargo-ok' a/Cargo.toml.orig b/Cargo.toml.orig
--- a/Cargo.toml.orig	2006-07-24 09:21:28.000000000 +0800
+++ b/Cargo.toml.orig	1970-01-01 08:00:00.000000000 +0800
@@ -1,35 +0,0 @@
-[package]
-name = "monero-ed25519"
-version = "0.1.0"
-description = "Ed25519 functionality, as within the Monero protocol"
-license = "MIT"
-repository = "https://github.com/monero-oxide/monero-oxide/tree/main/monero-oxide/ed25519"
-authors = ["Luke Parker <lukeparker5132@gmail.com>"]
-edition = "2021"
-rust-version = "1.65"
-
-[package.metadata.docs.rs]
-all-features = true
-rustdoc-args = ["--cfg", "docsrs"]
-
-[lints]
-workspace = true
-
-[dependencies]
-std-shims = { version = "0.1.5", default-features = false }
-
-subtle = { version = "2.5", default-features = false }
-zeroize = { version = "1.5", default-features = false, features = ["zeroize_derive", "alloc"] }
-
-rand_core = { version = "0.6", default-features = false }
-
-sha3 = { version = "0.10", default-features = false }
-
-crypto-bigint = { version = "0.5.1", default-features = false }
-curve25519-dalek = { version = "4", default-features = false, features = ["alloc", "zeroize"] }
-
-monero-io = { path = "../io", version = "0.1", default-features = false }
-
-[features]
-std = ["std-shims/std", "monero-io/std", "rand_core/std"]
-default = ["std"]
diff -ruN '--exclude=target' '--exclude=Cargo.lock' '--exclude=.cargo-checksum.json' '--exclude=.cargo-ok' a/.cargo_vcs_info.json b/.cargo_vcs_info.json
--- a/.cargo_vcs_info.json	2006-07-24 09:21:28.000000000 +0800
+++ b/.cargo_vcs_info.json	1970-01-01 08:00:00.000000000 +0800
@@ -1,6 +0,0 @@
-{
-  "git": {
-    "sha1": "9e11f5c0f2b18ab821192efa427c863086b51379"
-  },
-  "path_in_vcs": "monero-oxide/ed25519"
-}
\ 文件末尾没有换行符
diff -ruN '--exclude=target' '--exclude=Cargo.lock' '--exclude=.cargo-checksum.json' '--exclude=.cargo-ok' a/src/scalar.rs b/src/scalar.rs
--- a/src/scalar.rs	2006-07-24 09:21:28.000000000 +0800
+++ b/src/scalar.rs	2026-09-29 09:42:05.248001998 +0800
@@ -1,7 +1,6 @@
 use core::ops::DerefMut as _;
 
 #[allow(unused_imports)]
-use std_shims::prelude::*;
 use std_shims::io::{self, *};
 
 use subtle::{Choice, ConstantTimeEq, ConditionallySelectable};
